packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx

```diff
@@ -44,44 +44,54 @@ function addLabelDimension(
   }: Pick<ChartsXAxisProps, 'tickLabelInterval' | 'tickLabelStyle'> &
     Pick<AxisDefaultized, 'reverse'> & { isMounted: boolean },
 ): (TickItemType & LabelExtraData)[] {
-  const withDimension = xTicks.map((tick) => {
+  const getTickSize = (tick: TickItemType) => {
     if (!isMounted || tick.formattedValue === undefined) {
-      return { ...tick, width: 0, height: 0 };
+      return { width: 0, height: 0 };
     }
+
     const tickSizes = getWordsByLines({ style, needsComputation: true, text: tick.formattedValue });
+
     return {
-      ...tick,
       width: Math.max(...tickSizes.map((size) => size.width)),
       height: Math.max(tickSizes.length * tickSizes[0].height),
     };
-  });
+  };
 
-  if (typeof tickLabelInterval === 'function') {
-    return withDimension.map((item, index) => ({
-      ...item,
-      skipLabel: !tickLabelInterval(item.value, index),
-    }));
-  }
+  // FIXME: Add this back
+  // if (typeof tickLabelInterval === 'function') {
+  //  return withDimension.map((item, index) => ({
+  //    ...item,
+  //    skipLabel: !tickLabelInterval(item.value, index),
+  //  }));
+  // }
 
   // Filter label to avoid overlap
-  let currentTextLimit = 0;
   let previousTextLimit = 0;
   const direction = reverse ? -1 : 1;
-  return withDimension.map((item, labelIndex) => {
-    const { width, offset, labelOffset, height } = item;
+  const minGap = 8;
```

Comment:
Assumed a gap of 8px for testing purposes. Seems a reasonable default:

![image](https://github.com/user-attachments/assets/6e1109cb-3ee4-48ff-9792-c5cffb33ef50)